polar2grid/writers/hdf5.py

```diff
@@ -0,0 +1,74 @@
+import os
+import logging
+from satpy.writers import Writer
+from satpy import find_files_and_readers, Scene
+from datetime import datetime as datetime
+
+LOG = logging.getLogger(__name__)
+
+BASE_PATH = os.path.dirname(os.path.realpath(__file__))
+# FIXME: Use package_resources?
+PACKAGE_CONFIG_PATH = os.path.join(BASE_PATH, 'etc')
+
+config_fn = os.path.join(PACKAGE_CONFIG_PATH, 'hdf5.yaml')
+
+
+class hdf5writer(Writer):
+    """Writer for hdf5 files."""
+
+    # JMF NOTE:  figure out how to work with these, probably want to use some of these opts.
+    # some of these are dask some of these are pandas args for h5py options and not all of them
+    # make sense
+    H5PY_OPTIONS = ("compute",
+                    "lock",
+                    "scheduler",
+                    "key",
+                    "mode",
+                    "complevel",
+                    "complib",
+                    "append",
+                    "format",
+                    "errors",
+                    "encoding",
+                    "min_itemsize",
+                    "data_columns"
+                    )
+
+    def __init__(self, dtype=None, **kwargs):
+        """Init the writer."""
+        super(hdf5writer, self).__init__(default_config_filename=config_fn,
+                                            **kwargs)
+        self.dtype = self.info.get("dtype") if dtype is None else dtype
+
+        # h5py specific settings
+        self.h5py_options = {}
+        for k in self.H5PY_OPTIONS:
+            if k in kwargs or k in self.info:
+                self.h5py_options[k] = kwargs.get(k, self.info[k])
+
+    @classmethod
+    def separate_init_kwargs(cls, kwargs):
+        """Separate the init keyword args."""
+        # FUTURE: Don't pass Scene.save_datasets kwargs to init and here
+        init_kwargs, kwargs = super(hdf5writer, cls).separate_init_kwargs(
+            kwargs)
+        for kw in ['dtype']:
```

Comment:
I thought of this in the middle of the night last night...if you don't need to have arguments in the `__init__` then I wouldn't define any and would not define this `separate_init_kwargs` method. The `__init__` versus `__call__` argument handling is one of the biggest warts in Satpy's writers. If we can avoid dealing with it later (by not defining `__init__` args) then let's do it.